src/providers/npm.py

```diff
@@ -0,0 +1,82 @@
+from typing import List, Callable, Tuple
+import requests
+import urllib.request
+import os
+import multiprocessing
+import itertools
+from semver import max_satisfying
```

Comment:
Please add `requirements.txt`

Reply:
We should work `pipenv`, what do you think ?

Reply:
Why not conda?

Reply:
I will move to `conda`

Reply:
there is a problem `node-semver` is not published in conda

Reply:
https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-pkgs.html#installing-non-conda-packages

https://anaconda.org/conda-forge/python-node-semver

Reply:
let us wait for #13 to be resolved